internal/dataplane/sendconfig/strategy.go

```diff
@@ -12,10 +12,16 @@ import (
 	"github.com/kong/kubernetes-ingress-controller/v3/internal/metrics"
 )
 
+// CustomEntitiesByType stores all custom entities by types.
+// The key is the type of the entity,
+// and the corresponding slice stores the sorted list of custom entities with that type.
+type CustomEntitiesByType map[string][]map[string]any
```

Comment:
👍 

Nit: it's a new type, not an alias (and that's OK). Just wanted to make this clear that those are not identical.

Nit: we could consider making our own type for the custom entity itself instead of using `map[string]any` which is a bit opaque.

Something like.

```
type CustomEntity map[string]any
```

Those are loose comments/suggestions. Feel free to merge without addressing them.

Reply:
Because the type is used in here in package `sendconfig` and `GenerateSHA` in package `deckgen`, we cannot define a new type here. I used `custom.Object` definition from `go-kong` to substitute the  `map[string]any` here.

Reply:
Nothing precludes us from creating a separate package that will contain custom entity related types that would not have the dependency related issues (and could be then imported by both `sendconfig` and `deckgen`).

Reply:
If we really do this (extract ALL related types about custom entities to a separate place), this could make significant change to existing part of this PR. So it you DO think that is SO important, please create a new issue and finish that in a new PR.

Reply:
As mentioned in the original comment in this thread, that was just a suggestion which I believe would increase readability. Let's leave this as is and we can improve this later on if need be. 👍